### PR TITLE
Yet more ongoing cAVS cleanup & futureproofing

### PR DESCRIPTION
**Pull request by andyross**
Another batch of changes.  The big one being that the Xtensa double-mapped memory trick which underlies KERNEL_COHERENCE has been generalized and moved from the SOC into the Xtensa architecture, since it's something any hardware vendor might do and involves nothing specific to the audio DSPs at all.  And this allows us to configure it with just two kconfigs.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Comment by andyross** [comment]
Add one more patch to address a genuine bug I stumbled on (@dcpleung sorry, I dismissed your review, please check the new patch?) and remove the IPC header, which was dead code as it stands and which if things go well should re-appear as part of a proper driver in a day or two.

**Review by peter-mitsis — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

File: soc/xtensa/intel_adsp/common/include/cavs-link.ld [hunk 1]
```diff
@@ -129,7 +123,7 @@ MEMORY {
 	len = RAM_SIZE
 #ifdef CONFIG_KERNEL_COHERENCE
   ucram :
-	org = RAM_BASE - 0x20000000,
+	org = RPO_SET(RAM_BASE, CONFIG_XTENSA_UNCACHED_REGION),
```
Comment (teburd) [inline soc/xtensa/intel_adsp/common/include/cavs-link.ld:126]: This is a nice readability change

**Review by teburd — APPROVED** [review]
This seems like a nice changeset, I can appreciate the readability improvements in several places.

**Comment by marc-hb** [comment]
This does not make SOF PR testing worse https://github.com/thesofproject/sof/pull/5175

**Pull request merged**